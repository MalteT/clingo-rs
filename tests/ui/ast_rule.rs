use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let mut lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
    let hlit = ast::HeadLiteral::from(&lit);
    let mut blit = ast::BodyLiteral::from_literal(ast::Sign::NoSign, &lit);
    let body = vec![blit];
    let rule = ast::Rule::new(hlit, &body);
    blit = ast::BodyLiteral::from_literal(ast::Sign::NoSign, &lit);
    drop(body);
    lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
    let _end = (rule, lit, blit);
}