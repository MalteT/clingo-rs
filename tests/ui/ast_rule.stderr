error[E0505]: cannot move out of `body` because it is borrowed
  --> $DIR/ast_rule.rs:12:10
   |
10 |     let rule = ast::Rule::new(hlit, &body);
   |                                     ----- borrow of `body` occurs here
11 |     blit = ast::BodyLiteral::from_literal(ast::Sign::NoSign, &lit);
12 |     drop(body);
   |          ^^^^ move out of `body` occurs here
13 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
14 |     let _end = (rule, lit, blit);
   |                 ---- borrow later used here

error[E0506]: cannot assign to `lit` because it is borrowed
  --> $DIR/ast_rule.rs:13:5
   |
7  |     let hlit = ast::HeadLiteral::from(&lit);
   |                                       ---- borrow of `lit` occurs here
...
13 |     lit = ast::Literal::from_term(ast::Sign::NoSign, &term);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `lit` occurs here
14 |     let _end = (rule, lit, blit);
   |                 ---- borrow later used here

Some errors have detailed explanations: E0505, E0506.
For more information about an error, try `rustc --explain E0505`.
